onnx/helper.py

```diff
@@ -1098,3 +1095,121 @@ def make_training_info(
             binding.value = v
 
     return training_info
+
+
+# Following functions are used for mapping
+def tensor_dtype_to_np_dtype(tensor_dtype: int) -> np.dtype:
+    """
+    Convert a TensorProto's data_type to corresponding numpy dtype. It can be used while making tensor.
+
+    :param tensor_dtype: TensorProto's data_type
+    :return: numpy's data_type
+    """
+    return mapping.TENSOR_TYPE_MAP[int(tensor_dtype)].np_dtype
+
+
+def tensor_dtype_to_storage_tensor_dtype(tensor_dtype: int) -> int:
+    """
+    Convert a TensorProto's data_type to corresponding data_type for storage. It can be used for tensor_dtype_to_storage_numpy_type.
+
+    :param tensor_dtype: TensorProto's data_type
+    :return: data_type for storage
+    """
+    return mapping.TENSOR_TYPE_MAP[tensor_dtype].storage_dtype
+
+
+def tensor_dtype_to_string(tensor_dtype: int) -> str:
+    """
+    Get the name of given TensorProto's data_type.
+
+    :param tensor_dtype: TensorProto's data_type
+    :return: the name of data_type
+    """
+    return mapping.TENSOR_TYPE_MAP[int(tensor_dtype)].name
+
+
+def tensor_dtype_to_storage_numpy_type(tensor_dtype: int) -> np.dtype:
+    """
+    Convert a TensorProto's data_type to the data_type for storage.
+    Then further convert such a data_type for storage to corresponding numpy dtype. It can be used while to_array.
+
+    :param tensor_dtype: TensorProto's data_type
+    :return: numpy's data_type
+    """
+    return tensor_dtype_to_np_dtype(tensor_dtype_to_storage_tensor_dtype(tensor_dtype))
+
+
+def tensor_dtype_to_field(tensor_dtype: int) -> str:
+    """
+    Convert a TensorProto's data_type to corresponding field name for storage. It can be used while making tensors.
+
+    :param tensor_dtype: TensorProto's data_type
+    :return: field name
+    """
+    return cast(
+        str,
+        mapping.STORAGE_TENSOR_TYPE_TO_FIELD[
+            mapping.TENSOR_TYPE_MAP[int(tensor_dtype)].storage_dtype
+        ],
+    )
+
+
+def np_dtype_to_tensor_dtype(np_dtype: np.dtype) -> int:
+    """
+    Convert a numpy's dtype to corresponding tensor type. It can be used while converting numpy arrays to tensors.
+
+    :param np_dtype: numpy's data_type
+    :return: TensorsProto's data_type
+    """
+    return cast(int, mapping.NP_TYPE_TO_TENSOR_TYPE[np_dtype])
+
+
+def sequence_type_to_field(elem_type: int) -> str:
+    """
+    Convert a SequenceProto's elem_type to corresponding field name.
+
+    :param tensor_dtype: SequenceProto's elem_type
+    :return: field name
+    """
+    return cast(str, mapping.STORAGE_ELEMENT_TYPE_TO_FIELD[elem_type])
+
+
+def get_attr_from_sequence_elem_type(tensor: SequenceProto, elem_type: int) -> Any:
```

Comment:
I don't think we need these utility/helper methods. Please see my comment in `numpy_helper.py` on rewriting the code there. I think splitting the code into these two files actually makes the logic of either one harder to understand. (I understand this is a problem with the pre-existing code introduced by someone else previously, and not this PR.)

Reply:
For readability, yes I think keeping if-else statement makes more sense, but for maintainability, having a common function for the same mapping seems not too bad (we don't need to change the mapping in several places if mapping is changed). However, if the attribute mapping for protos is quite stable, I am fine with having if-else statements. I have removed attribute map for SequenceProto and OptionalProto and used if-else statements instead.

Not sure whether it is a concern -- that also means that previously the attribute mappings for SequenceProto and OptionalProto are public used functions and in the future they will be removed.